src/main/java/com/contentgrid/gateway/runtime/routing/DefaultRuntimeRequestRouter.java

```diff
@@ -0,0 +1,38 @@
+package com.contentgrid.gateway.runtime.routing;
+
+import com.contentgrid.gateway.runtime.application.ServiceCatalog;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.cloud.client.ServiceInstance;
+import org.springframework.web.server.ServerWebExchange;
+import reactor.core.publisher.Mono;
+
+@Slf4j
+@RequiredArgsConstructor
+public class DefaultRuntimeRequestRouter implements RuntimeRequestRouter {
+
+    private final ServiceCatalog serviceCatalog;
+    private final RuntimeVirtualHostResolver virtualHostResolver;
+    private final RuntimeServiceInstanceSelector serviceInstanceSelector;
+
+    @Override
+    public Mono<ServiceInstance> route(ServerWebExchange exchange) {
+        return Mono.just(exchange.getRequest().getURI())
+                .map(this.virtualHostResolver::resolve)
+                .flatMap(Mono::justOrEmpty)
+                .switchIfEmpty(Mono.defer(() -> {
+                    log.debug("Could not resolve {} to app-id", exchange.getRequest().getURI());
```

Comment:
Are you doing to put the full request URI in here? Wouldn't it be better to only put the hostname in? Full URIs get quite long, and are quite useless for our mapping case as well.

Reply:
My intention was to put the full URI here to make it easier to match against access logs.

But thinking this over: that is the purpose of the correlation-id we should have in the json-logs, so I'll add `.getHost()` to it

Reply:
Well, are you going to put `getHost()` in here?